tests/testthat/test-precompile.R

```diff
@@ -0,0 +1,83 @@
+# Disable caching
+sass_cache_set(NULL)
+
+describe("Can find precompiled themes", {
+  # Should be present for 4, 4+3, 3 without bootswatch theme
+  expect_true(file.exists(precompiled_css_path(bs_theme_create("4"))))
+  expect_true(file.exists(precompiled_css_path(bs_theme_create("4+3"))))
+  expect_true(file.exists(precompiled_css_path(bs_theme_create("3"))))
+
+  # Not present for bootswatch themes
+  expect_null(precompiled_css_path(bs_theme_create("4", "sketchy")))
+  expect_null(precompiled_css_path(bs_theme_create("4+3", "darkly")))
+
+  # With modifications
+  bs_theme_new()
+  bs_theme_add("body { background: red; }")
+  expect_null(precompiled_css_path(bs_theme_get()))
+})
+
+
+describe("Precompiled theme output is identical to compiled themes", {
+  get_bootstrap_path <- function(x) {
+    bs_dep <- Filter(x, f = function(y) { identical(y$name, "bootstrap") })[[1]]
+    bs_dep$src$file
+  }
+
+  # precompiled themes will end up in a different directory, but have identical
+  # contents to non-precompiled. The dirs will be something like:
+  # /tmp/RtmpVi9qRT/bootstraplib-088b402ee16511d7
+  # /tmp/RtmpVi9qRT/bootstraplib-precompiled-4
+  dir1 <- get_bootstrap_path(bootstrap(bs_theme_create("4"), use_precompiled_css = FALSE))
+  dir2 <- get_bootstrap_path(bootstrap(bs_theme_create("4"), use_precompiled_css = TRUE))
+  expect_false(identical(dir1, dir2))
+  expect_true(identical_dirs(dir1, dir2))
+
+  dir1 <- get_bootstrap_path(bootstrap(bs_theme_create("4+3"), use_precompiled_css = FALSE))
+  dir2 <- get_bootstrap_path(bootstrap(bs_theme_create("4+3"), use_precompiled_css = TRUE))
+  expect_false(identical(dir1, dir2))
+  expect_true(identical_dirs(dir1, dir2))
+
+  dir1 <- get_bootstrap_path(bootstrap(bs_theme_create("3"), use_precompiled_css = FALSE))
+  dir2 <- get_bootstrap_path(bootstrap(bs_theme_create("3"), use_precompiled_css = TRUE))
+  expect_false(identical(dir1, dir2))
+  expect_true(identical_dirs(dir1, dir2))
```

Comment:
```suggestion
  for (version in c("3", "4", "4+3")) {
    dir1 <- get_bootstrap_path(bootstrap(bs_theme_create(version), use_precompiled_css = FALSE))
    dir2 <- get_bootstrap_path(bootstrap(bs_theme_create(version), use_precompiled_css = TRUE))
    expect_false(identical(dir1, dir2))
    expect_true(identical_dirs(dir1, dir2))
  }
  
```

Reply:
I generally try to avoid having test expectations in functions and loops, because when they do fail, it can be difficult to figure out which condition failed.

For example, the output of testthat looks like this if one of these tests fails:

```
test-precompile.R:39: failure: Precompiled theme output is identical to compiled themes
identical_dirs(dir1, dir2) isn't true.
```

If it's in a loop, we don't know which condition had a problem.